### PR TITLE
feat: Add Project version info

### PR DESCRIPTION
**Pull request by JacopoMangiavacchi**
- Add Project version to project file
- Add Project version to asset metadata file
- Show Project version in App Settings Page

Implement [#AB17100]

### CODE REVIEW

File: src/models/applicationState.ts [hunk 1]
```diff
@@ -98,6 +98,7 @@ export interface IAppSettings {
 export interface IProject {
```
Comment (wbreza) [inline src/models/applicationState.ts]: What do you think about defining the `version` as optional - So it makes defining the objects easier from the consumer side and then the services/reducers job is to set it when it changes.

File: src/services/assetService.test.ts [hunk 2]
```diff
@@ -150,6 +151,7 @@ describe("Asset Service", () => {
                     state: AssetState.Tagged,
                 },
                 regions: [],
+                version: "",
             };
```
Comment (wbreza) [inline src/services/assetService.test.ts]: I would expect this test to fail right now - If we are setting version to empty string the `assetService.save(...)` should be applying the version from the package.json therefore making the equality comparison fail below.

File: src/redux/actions/projectActions.ts [hunk 3]
```diff
@@ -70,7 +71,9 @@ export function saveProject(project: IProject)
             throw new AppError(ErrorCode.SecurityTokenNotFound, "Security Token Not Found");
         }
 
-        const savedProject = await projectService.save(project, projectToken);
+        const newProject = {...project, version: packageJson.version};
```
Comment (wbreza) [inline src/redux/actions/projectActions.ts]: Let's make sure we have a new test the validates the version is set when `saveProject` is called.

File: src/models/applicationState.ts (hunk 1 above)
Comment (JacopoMangiavacchi) [reply src/models/applicationState.ts]: It's "optional" to read of course but I think it should be mandatory to always persist project (and asset) with this new value.  If we mark this optional we're not guaranteed.

File: src/services/assetService.test.ts (hunk 2 above)
Comment (JacopoMangiavacchi) [reply src/services/assetService.test.ts]: Not really. The assetService.save() method just save an asset metadata it receives in input.  In that specific test the goal was just to check this method that don't really care about the value of asset metadata. The thing that that property is not optional just enforce that at least it as some value and an empty string is still a value.   Btw real project version value is enforced in the redux.  All the test that use the redux save project has been updated in this PR asserting for the valid project version.

File: src/redux/actions/projectActions.ts (hunk 3 above)
Comment (JacopoMangiavacchi) [reply src/redux/actions/projectActions.ts]: Added test for both saveProject and saveAssetMetadata

File: src/react/components/pages/appSettings/appSettingsPage.tsx [hunk 4]
```diff
@@ -60,6 +61,9 @@ export default class AppSettingsPage extends React.Component<IAppSettingsProps>
                     onSubmit={this.onFormSubmit}
                     onCancel={this.onFormCancel} />
                 <div className="app-settings-page-sidebar p-3 bg-lighter-1">
+                    <div className="my-3">
+                        <p>{strings.appSettings.version.description + packageJson.version}</p>
```
Comment (wbreza) [inline src/react/components/pages/appSettings/appSettingsPage.tsx]: Consider using string interpolation vs appending string.

**Pull request merged**